shong.wang/2019/05/fifa.tsml

```diff
@@ -0,0 +1,994 @@
+                        <ts:value xml:lang="es">Estadio de Nizhni Novgorod</ts:value>
+                    </ts:option>
+                    <ts:option key="7">
+                        <ts:value xml:lang="ru">Стадион Калининград</ts:value>
+                        <ts:value xml:lang="en">Luzhniki Stadium</ts:value>
+                        <ts:value xml:lang="zh">卢日尼基体育场</ts:value>
+                        <ts:value xml:lang="es">Estadio Luzhniki</ts:value>
+                    </ts:option>
+                    <ts:option key="8">
+                        <ts:value xml:lang="ru">Стадион Лужники</ts:value>
+                        <ts:value xml:lang="en">Samara Arena</ts:value>
+                        <ts:value xml:lang="zh">萨马拉体育场</ts:value>
+                        <ts:value xml:lang="es">Samara Arena</ts:value>
+                    </ts:option>
+                    <ts:option key="9">
+                        <ts:value xml:lang="ru">Стадион Нижний Новгород</ts:value>
+                        <ts:value xml:lang="en">Rostov Arena</ts:value>
+                        <ts:value xml:lang="zh">罗斯托夫体育场</ts:value>
+                        <ts:value xml:lang="es">Rostov Arena</ts:value>
+                    </ts:option>
+                    <ts:option key="10">
+                        <ts:value xml:lang="ru">Стадион Спартак</ts:value>
+                        <ts:value xml:lang="en">Spartak Stadium</ts:value>
+                        <ts:value xml:lang="zh">斯巴达克体育场</ts:value>
+                        <ts:value xml:lang="es">Estadio del Spartak</ts:value>
+                    </ts:option>
+                    <ts:option key="11">
+                        <ts:value xml:lang="ru">Стадион Санкт-Петербург</ts:value>
+                        <ts:value xml:lang="en">Saint Petersburg Stadium</ts:value>
+                        <ts:value xml:lang="zh">圣彼得堡体育场</ts:value>
+                        <ts:value xml:lang="es">Estadio de San Petersburgo</ts:value>
+                    </ts:option>
+                    <ts:option key="12">
+                        <ts:value xml:lang="ru">Стадион Фишт</ts:value>
+                        <ts:value xml:lang="en">Mordovia Arena</ts:value>
+                        <ts:value xml:lang="zh">莫多维亚体育场</ts:value>
+                        <ts:value xml:lang="es">Mordovia Arena</ts:value>
+                    </ts:option>
+                </ts:mapping>
+            </ts:token-id>
+        </ts:origins>
+    </ts:attribute-type>
+
+    <ts:attribute-type id="time" syntax="1.3.6.1.4.1.1466.115.121.1.24">
+        <ts:name>
+            <ts:string xml:lang="en">Time</ts:string>
+            <ts:string xml:lang="zh">时间</ts:string>
+            <ts:string xml:lang="es">Tiempo</ts:string>
+            <ts:string xml:lang="ru">время</ts:string>
+        </ts:name>
+        <ts:origins>
+            <ts:token-id as="utf8" bitmask="000000000000000000000000000000000000FFFFFFFF00000000000000000000"/>
```

Comment:
This should be `as="uint"` instead:

```
<ts:token-id as="utf8" bitmask="000000000000000000000000000000000000FFFFFFFF00000000000000000000"/>
```

And we need a mapping for the uint to generalised time.